File: acme-java-reactive/src/main/java/acme/dto/TransactionResponse.java
package acme.dto;

import java.time.LocalDateTime;
import java.util.UUID;

public record TransactionResponse(
    UUID transactionId,
    UUID fromAccountId,
    UUID toAccountId,
    Long amount,
    LocalDateTime timestamp,
    String status
) {
    public TransactionResponse(UUID transactionId, UUID fromAccountId, UUID toAccountId, Long amount) {
        this(transactionId, fromAccountId, toAccountId, amount, LocalDateTime.now(), "SUCCESS");
    }
}

File: acme-springboot-nonblocking/src/main/kotlin/acme/dto/TransactionResponse.kt
package acme.dto

import java.time.LocalDateTime
import java.util.*

data class TransactionResponse(
    val transactionId: UUID = UUID.randomUUID(),
    val fromAccountId: UUID,
    val toAccountId: UUID,
    val amount: Long,
    val timestamp: LocalDateTime = LocalDateTime.now(),
    val status: String = "SUCCESS"
)
